pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -281,7 +281,16 @@ func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec,
 }
 
 func (p *provider) GetCloudConfig(spec v1alpha1.MachineSpec) (config string, name string, err error) {
-	return "", "", nil
+	c, _, err := p.getConfig(spec.ProviderSpec)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to parse config: %v", err)
+	}
+	cc := kubevirttypes.CloudConfig{
+		Kubeconfig: c.Kubeconfig,
+	}
+	ccs, err := cc.String()
+
+	return ccs, string(providerconfigtypes.CloudProviderKubeVirt), err
```

Comment:
Ok sounds like this is not gonna work, please remove the cloud provider from here as KubeVirt is not a known cloud provider to Kubelet and change that with `external` . At least in the k8s versions that we support

Reply:
done